x/ecosystem-incentive/keeper/nftmarket_frontend_service.go

```diff
@@ -0,0 +1,188 @@
+// The implementations especially for the services about the Nftmarket Frontend reward.
+// The reason why it's separated is for achieving the explicity and extensibility of this module.
+
+package keeper
+
+import (
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	"github.com/UnUniFi/chain/x/ecosystem-incentive/types"
+	nftmarkettypes "github.com/UnUniFi/chain/x/nftmarket/types"
+)
+
+// RecordIncentiveIdWithNftId is for recording incentiveUnitId with nftId
+// to know of the receriver of the incentive reward for the frontend creator
+// of Nftmarket in AfterNftPaymentWithCommission method.
+func (k Keeper) RecordIncentiveUnitIdWithNftId(ctx sdk.Context, nftId nftmarkettypes.NftIdentifier, incentiveUnitId string) {
+	// panic if the nftId is already recorded in the store.
+	if _, exists := k.GetIncentiveUnitIdByNftId(ctx, nftId); exists {
+		panic(sdkerrors.Wrap(types.ErrRecordedNftId, nftId.String()))
+	}
+
+	// check incentiveUnitId is already registered
+	if _, exists := k.GetIncentiveUnit(ctx, incentiveUnitId); !exists {
+		_ = fmt.Errorf(sdkerrors.Wrap(types.ErrNotRegisteredIncentiveUnitId, incentiveUnitId).Error())
```

Comment:
This one is not used I think - where it's being used?

Reply:
Do you mean this condition?
if so, it checks if incentive in a memo field of nft listing tx is already registered by sending a msg to record with nft id to accumulate reward later on.

Reply:
now i get what you mean.
what do you suggest instead of `fmt.Errorf`?

Reply:
do you think it has to be logged?
or simply remove and done?